h/util/group_scope.py

```diff
@@ -18,25 +18,69 @@ def match(uri, scopes):
     return scope in scopes
 
 
+def uri_in_scope(uri, scopes):
+    """
+    Does the URI match any of the scope patterns?
+
+    Return True if the URI matches one or more patterns in scopes (if the
+    URI string begins with any of the scope strings)
+
+    :arg uri: URI string in question
+    :arg scopes: List of URIs that define scope
+    :type scopes: list(str)
+    :rtype: bool
+    """
+    return any((uri.startswith(scope) for scope in scopes))
+
+
+# TODO: This concept no longer makes sense with more granular scoping. There is
+# no equivalent 1:1 uri <-> scope relationship. Remove this function soon.
 def uri_scope(uri):
     """
     Return the scope for a given URI
 
     Parse a scope from a URI string. Presently a scope is an origin, so this
-    proxies to _parse_origin.
+    proxies to parse_origin.
     """
-    return _parse_origin(uri)
+    return parse_origin(uri)
+
 
+def uri_to_scope(uri):
```

Comment:
I think a naming pass is due in a follow-up PR, to wit:

* Normalise usage of `uri` vs `url`
* Stop overloading the term `scope`, as noted here

I'm going to try to avoid putting any additional logic in the model, as @seanh has set a pattern that we should keep the models as thin as possible.